### ISSUE TITLE
FEATURE: Set SAM's to "random" behaviour

### ISSUE THREAD

**Issue by 132ndNeck** [132nd-vWing/OPBR-Brief#10]
Request is to set certain SAM groups to operate in "random" behaviour  (As opposed to the "trap" mode in Issue #8 )

In random mode the SAM group behaves in the following manner:
- Randomly turn off/on the radar ( on for 60-180 seconds, off for 120-360 seconds)
- When off, moves to a randomly new location 20-100 meters away from current location  (THIS ONLY FOR SAM's that can move, not SA-3, SA-2, SA-6, SA-10).
- If something is detected within the WEZ (weapon engagement zone) when it is on,  the SAM stays on and tries to shoot down the aircrafts

**Comment by thebgpikester** [comment]
first two points doable fairly simply by having these named with a prefix and have that script run a timer and flip the switches for move, nothing and illuminate.

The last one is the more challenging though. It's probably easier to abstract if any blue are in a certain range then switch on.
Just SA6, SA8m SA11, SA15, SA19 named mobile-SAM-x
mobiles = SET_GROUP:New():FilterPrefixes("mobile-SAM-"):FilterStart()
Schedule
1. For each SAM
2. Any blue in range of me? If so, ROEWeaponsGreen()
3. Roll dice. 50% on or 50% 
4. If on, stays on
5. If off, moves ROEWeaponsRed() RouteToVec2(random point in small zone)
Repeat every 60-180 seconds

wont be entirely accurate to the specification but that's the jist

**Comment by 132ndNeck** [comment]
Would this be resource intensive for the server?

**Comment by thebgpikester** [comment]
probably OK, unless lots of them moving at once. But dont expect lots of sams doing this

**Comment by 132ndNeck** [comment]
No, do not expect a lot of SAM groups to do this, but to have the ability, so that the players do not know 100% what to anticipate from the IADS, making it more lethal
